Modifications to show assurance cases in Eclipse Views

davidcok opened the pull request.

tuxji commented on tools/rack/rack.plugin/META-INF/MANIFEST.MF:
```diff
@@ -42,7 +42,8 @@ Require-Bundle: com.ge.research.jena,
  org.yaml.snakeyaml,
- org.eclipse.debug.ui
+ org.eclipse.debug.ui,
+ openjfx.swing;bundle-version="17.0.2"
```
I'm surprised this requirement doesn't cause any error.  We are explicitly adding the javafx jars to this bundle's classpath instead of requiring Eclipse to find the javafx bundles in its P2 repositories.  At best, this requirement is unnecessary; at worst, it may be muddling where the swing classes come from (are they coming from a bundle in P2 or a jar in the classpath?).  We should remove this requirement since RITE still should work without it.

davidcok replied:
Reverted the change. I suspect Eclipse inserted it.

tuxji commented on tools/rack/rack.plugin/META-INF/MANIFEST.MF:
```diff
@@ -90,7 +91,11 @@ Bundle-ClassPath: .,
+ lib/javafx-swing.jar,
+ lib/javafx-swing-linux.jar,
+ lib/javafx-swing-mac.jar,
+ lib/javafx-swing-win.jar
```
Please keep all the jars sorted alphabetically.  This helps maintainers check for missing jars which are in the lib directory but not on the bundle's classpath (note we do omit some jars from the lib directory on purpose, but still).

davidcok replied:
Fixed.

tuxji commented on tools/rack/rack.plugin/build.properties:
```diff
@@ -53,7 +53,11 @@ bin.includes = .,\
+               lib/javafx-swing.jar,\
+               lib/javafx-swing-linux.jar,\
+               lib/javafx-swing-mac.jar,\
+               lib/javafx-swing-win.jar
```
Likewise, please keep these filenames sorted alphabetically so maintainers can check for missing files more easily.

davidcok replied:
Fixed.

tuxji commented on tools/rack/rack.plugin/pom.xml:
```diff
@@ -42,115 +42,162 @@
     <dependency>
       <groupId>org.openjfx</groupId>
       <artifactId>javafx-base</artifactId>
+      <version>${version.javafx}</version>
```
All of these newly added versions are redundant because the parent pom already has the complete information (including versions) in its dependencyManagement section.  This allows the child pom to "inherit" the version from the parent pom.

davidcok replied:
Removed them.

davidcok commented on tools/rack/pom.xml:
```diff
@@ -295,9 +318,7 @@
-              <goals>
-                <goal>apply</goal>
-              </goals>
+              <goals></goals>
```
Oops - a local change snuck into the commit (this task does not work in my environment).  Reverted.

davidcok commented:
1. Removed the (debugging) "Mode" dialog
2. As expected
3. The flickering comes because the SADL Console grabs focus whenever logging output is written to it. I minimized that focus grabbing for the RACK console, but the work needs to be done also for the SADL console (which I don't have access to). Otherwise - I don't have a windows environment to assess the other comments. I have not observed them on Mac, but given a test scenario would be willing to try it.

Updated with a merge from main and pushed changes to the PR

pauls4GE commented:
Need to test on windows before merging
